src/main/CollectorModule.ts

```diff
@@ -68,17 +69,28 @@ export class CollectorModule {
 		try {
 			collector.attach(writer, log);
 		} catch (e) {
-			log.error("Unexpected Exception during collector attach: ", e);
+			log.error(`[${collector.constructor.name}] Unexpected Exception during collector attach: `, e);
 		}
 	}
 
 	private getLogger() {
-		return this.logger || new TransportLogger(this.transports);
+		const hasLogger = !!this.logger;
+		if (hasLogger)
+			return this.logger;
+
+		if (!this.transports || this.transports.length === 0) {
+			console.warn("ATTENTION-SEARCH-COLLECTOR-WARNING");
+			console.warn("search-collector: no LoggerTransport configured while using the default TransportLogger. Please add a transport CollectorModule#addLogTransport or CollectorModule#setTransports");
+			console.warn("search-collector: will FALLBACK to ConsoleTransport");
+			return new TransportLogger([new ConsoleTransport()]);
+		}
```

Comment:
Why not just make the console transport the default i.e. what do we gain with the warning?

Reply:
mh, good point. Initially I thought people should definetly override the transport, thats why I'm spamming the user with these log messages